From 634c04efc904225a87794f3b048147098a409da2 Mon Sep 17 00:00:00 2001
From: Jason Montleon <jmontleo@redhat.com>
Date: Thu, 19 Mar 2026 12:43:00 -0400
Subject: [PATCH 0001/1239] Revert "net: spacemit: Fix error handling in
 emac_tx_mem_map()"

This reverts commit edeaba385318f60ec1b32470da4d5eb800294d16.
---
 drivers/net/ethernet/spacemit/k1_emac.c | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/drivers/net/ethernet/spacemit/k1_emac.c b/drivers/net/ethernet/spacemit/k1_emac.c
index d64ca7bbda9e..5de69a105168 100644
--- a/drivers/net/ethernet/spacemit/k1_emac.c
+++ b/drivers/net/ethernet/spacemit/k1_emac.c
@@ -750,7 +750,7 @@ static void emac_tx_mem_map(struct emac_priv *priv, struct sk_buff *skb)
 	struct emac_desc tx_desc, *tx_desc_addr;
 	struct device *dev = &priv->pdev->dev;
 	struct emac_tx_desc_buffer *tx_buf;
-	u32 head, old_head, frag_num, f, i;
+	u32 head, old_head, frag_num, f;
 	bool buf_idx;
 
 	frag_num = skb_shinfo(skb)->nr_frags;
@@ -818,15 +818,6 @@ static void emac_tx_mem_map(struct emac_priv *priv, struct sk_buff *skb)
 
 err_free_skb:
 	dev_dstats_tx_dropped(priv->ndev);
-
-	i = old_head;
-	while (i != head) {
-		emac_free_tx_buf(priv, i);
-
-		if (++i == tx_ring->total_cnt)
-			i = 0;
-	}
-
 	dev_kfree_skb_any(skb);
 }
 
-- 
2.53.0

